compiler-core/src/language_server/server.rs

```
@@ -94,9 +103,37 @@ where
         match message {
             Message::Request(id, request) => self.handle_request(id, request),
             Message::Notification(notification) => self.handle_notification(notification),
+            Message::Response(response) => self.handle_response(response),
+        }
+    }
+
+    fn handle_response(&mut self, response: Response) {
+        match response {
+            Response::Configuration(updated_config) => {
+                {
+                    let mut config = self.config.write().expect("cannot write config");
+                    *config = updated_config;
+                }
+
+                self.inlay_hints_refresh();
+            }
         }
     }
 
+    fn send_request(
+        &mut self,
+        method: &str,
+        params: impl serde::Serialize,
+        handler: Option<ResponseHandler>,
+    ) {
+        let request = self.message_buffer.make_request(method, params, handler);
+
+        self.connection
+            .sender
+            .send(lsp_server::Message::Request(request))
+            .unwrap_or_else(|_| panic!("send {method}"));
```

Do not discard the error please, otherwise it's hard to debug 🙏